jbmc/unit/java-testing-utils/require_goto_statements.cpp

```diff
@@ -295,6 +294,66 @@ const code_declt &require_goto_statements::require_declaration_of_name(
   throw no_decl_found_exceptiont(variable_name.c_str());
 }
 
+/// Get the unique non-null expression assigned to a symbol. The symbol may have
+/// many null assignments, but only one non-null assignment.
+/// \param entry_point_instructions: A vector of instructions
+/// \param symbol_identifier: The identifier of the symbol we are considering
+/// \return The unique non-null expression assigned to the symbol
+const exprt &get_unique_non_null_expression_assigned_to_symbol(
+  const std::vector<codet> &entry_point_instructions,
+  const irep_idt &symbol_identifier)
+{
+  const auto &assignments = require_goto_statements::find_pointer_assignments(
+                              symbol_identifier, entry_point_instructions)
+                              .non_null_assignments;
+  REQUIRE(assignments.size() == 1);
+  return assignments[0].rhs();
+}
+
+/// Get the unique symbol assigned to a symbol, if one exists. There is always
+/// a unique non-null assignment to the symbol, and it is either another symbol
+/// or a dynamic allocation.
```

Comment:
This doesn't enforce that it's a dynamic allocation, so just say it's either copied from another symbol (in which case we return that symbol expression) or from something else (when we return null)